### DISCUSSION TITLE
GLPI 10.0.0 and GLPI-agent 1.5 no licenses inventory

### DISCUSSION THREAD

**Discussion by mareban** [glpi-project/glpi-agent#513] [Q&A]
### Bug reporting acknowledgment

Yes, I read it

### Professional support

None

### Describe the bug

Hi,

Is software licenses inventory  still supported, or is this a bug or a misconfiguration (some paremeters to set, dependancy issues, ...) on our side ? 

Thanks for your help.

### To reproduce

glpi-agent inventory

### Expected behavior

get software licenses

### Operating system

Linux

### GLPI Agent version

1.5

### GLPI version

10.0.9

### Additional context

nope

**Comment by g-bougard** [comment]
Hello @mareban 

I converted your issue report as a Q&A discussion as you're really asking a question and not reporting an issue.

Software license inventory is supported for some softwares on windows and MacOSX: Microsoft Office, some Adobe or VMWare softwares for example. On windows, the agent tries to report all licences registered in WMI objects.

If you think a license is missing, you should open a feature request or an idea discussion. If this concerns a still supported software, we of course may have a bug somewhere or maybe the license data format has changed and we may need to update the way it is collected.

**Comment by mareban** [comment]
Hello,

So sorry, and thanks for your reply.

We'll try to understand why we don't have any licenses  in our infrastructure on more than 1200 assets in just one site from others around the world, mainly windows machines with Microsoft Office, Adobe,  ... , because we don't have any in management/licenses ...

BR

**Reply by g-bougard** [reply]
Hi @mareban 

to investigate, you can take a platform, check what licenses you're expecting and run the following command from and administrator console and from the glpi-agent installation folder:
```
glpi-inventory --partial licenseinfo --debug --debug
```
You can redirect standard output toward a file to view the generated partial inventory from a text editor. Then you can check the output for any error related to License module and you can look for a LICENSEINFOS node in the inventory file.

**Comment by mareban** [comment]
Hello,

Thanks again for your help :-) ! Yes, we have for my Lenovo laptop  a licenseinfos node, with 2 office16365ProPlus... in the output, with no error messages but nothing from the web console !

We don't have any licenses reported and we should licence keys from a lot  of product tools used internally on this site ?

Maybe some parameters to activate, configuration or permissions issue  ?


BR

**Reply by g-bougard** [reply]
Hi @mareban 

it seems to me you're talking about 2 problems:
 1. you have Office licenses datas, but it seems not reported in GLPI
 2. you expect a lot of more license keys reported

For the first point, I can't say anything without datas to try. You can eventually check on your GLPI server if you found some error in the php-errors.log file that triggers during the inventory import. The best would be to share the output of glpi-inventory script as I would be able to try it in a test instance.

For the second point, I can't say anything if you don't even list all the software licenses you expect to be reported. You can eventually share the output of `wmic path SoftwareLicensingProduct get /format:list` if something in this output should be included in the LICENSEINFOS node but is not.

To answer your question, by default, glpi-agent reports licenses infos it discovers. Most softwares store the license info in a private place, so they may require some development if the agent does support them currently.

If you're a company, you should probably look for a [GLPI Network subscription](https://services.glpi-network.com/) so you can report you're problem privately to us as editor. Of course, if we develop dedicated modules for the agent, they would be released as opensource.

**Reply by mareban** [reply]
Hi ,


Thanks for your reply ! With the wmic command , i've a lot of Microsoft 0365 stuff , but i don't know if it helps ?

I've seen this issue too, maybe it's related : https://github.com/glpi-project/glpi-agent/issues/323 ?

But again, we've a lot of commercial applications installed in this site, and no license in the inventory :-( !

![image](https://github.com/glpi-project/glpi-agent/assets/11810873/a0873697-8ed0-481b-9213-1f84f5a5cb84)


Thanks again for your support 

![image](https://github.com/glpi-project/glpi-agent/assets/11810873/2396ad8e-af1d-4f49-94e6-82479109600c)

**Reply by g-bougard** [reply]
Hi,

#323 fix is included in glpi-agent 1.5. So you should not be concerned.

> a lot of Microsoft 0365 stuff

Do you need them to be inventoried ? I really don't know if we really can inventory something.

There's a lot of entries in wmic output, but of course, glpi-agent filters them to only keep the one we think are required. If you see few that needs to be included, please share the relevant part from wmic output so I can check if it's possible.

> But again, we've a lot of commercial applications installed in this site, and no license in the inventory :-( !

We can't do anything if you just complain and don't explain what you expect: how many softwares, softwares name and version ? Do you know a method to get the license numbers ?
As I said, you should really consider to buy a GLPI subscription and ask us to add new software license support in glpi-agent.

**Comment by kintaro1981** [comment]
GLPI Agent 1.5
glpi-inventory 1.2.3
glpi 10.0.9

GLPI is not displaying office licenses for me too.
I can see them in the XML.
(or maybe I don't know where to look?)

[NB-COMPUTERNAME-2023-09-21-15-33-14.zip](https://github.com/glpi-project/glpi-agent/files/12893094/NB-COMPUTERNAME-2023-09-21-15-33-14.zip)
(I can't upload xml, I zipped it)

**Reply by g-bougard** [reply]
I can't say if it's a glpi-agent issue or GLPI one without the content (it seems we can't download the XML).
Anyway do you see some LICENSEINFOS nodes related to Office in your XML ?

**Reply by kintaro1981** [reply]
I fixed the link, try again to download the zip file

**Reply by g-bougard** [reply]
Okay, thanks.

LICENSEINFOS nodes are there. But it seems they are not supported in native inventory. Sorry for that, I wasn't aware of any problem regarding this.

To me this should like a server issue so you should open an issue on GLPI project and link it to this discussion.
